lib/loglevel.js

```diff
@@ -261,18 +307,33 @@
 
     defaultLogger = new Logger();
 
-    defaultLogger.getLogger = function getLogger(name) {
-        if ((typeof name !== "symbol" && typeof name !== "string") || name === "") {
-            throw new TypeError("You must supply a name when creating a logger.");
+    defaultLogger.getLogger = function getLogger(_singleName) {
+        if( !arguments.length ) {
```

Comment:
For consistency and simplicity elsewhere, it'll actually be nicer to allow this, such that `logger.getLogger()` returns `logger`. It'll make the model & code cleaner to avoid the special case, and means that things like `defaultLogger.getLogger(keyHierarchy.slice(0, -1))` work correctly for the first level of the hierarchy.